internal/mutators/podtemplatespec_test.go

```diff
@@ -0,0 +1,362 @@
+package mutators
+
+import (
+	"testing"
+
+	"github.com/kanopy-platform/hedgetrimmer/internal/limitrange"
+	"github.com/kanopy-platform/hedgetrimmer/internal/quantity"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestApplyResourceRequirements(t *testing.T) {
```

Comment:
I think the function is testing too much.  There are two table tests, with one that is testing 3 - 5 different mutations on the different containers.

I think the table tests should be split into their own functions and own table tests.

The matrix should examine each mutation case and container.

For each
 - InitContainer
 - Container

Have Tests:

- TestMutationWithNoRequestsAndLimits.
- TestMutationWithRequestsAndNoLimits
- TestMutationWithMemoryRequestAndNoLimits

Are there other cases missing?

Reply:
Yeah agreed this test was messy. I tried to split it into the matrix as you described but it had its issues:
- Testing InitContainer and Container separately in each test is awkward because for comparison I need to know which to round up
- Both InitContainer and Container are just a collection of containers so they are really testing the same thing and each test would repeat for both which seems annoying

Instead I cleaned up the test to only specify the containers as input which is what we're after. All the test cases are in the table test. The test logic then constructs InitContainer and Container flavors separately and runs tests against them. This way introducing a new test ensures it gets tested on both automatically.

Reply:
It is much easier to parse now. Thanks!